wasi/wasi.go

```diff
@@ -28,9 +28,12 @@ const (
 	R_FD_WRITE
 )
 
+// File is an interface for an opened file similar to fs.File, but it supports addtional operations such as Write.
```

Comment:
Change this entirely as the it is a side-note that this happens to match another thing

```
File combines file I/O interfaces supported by ModuleSnapshotPreview1
```

os.File isn't important and we shouldn't mention it. The more important type is fs.File (because we want to end up with an impl of `fs.FS`)

IMHO, we shouldn't mention fs.File until we finish the things it supports. Then we embed the fs.File interface along with anything extra we support.

Ex. you'll notice we don't have stat here, yet.
```
// A File provides access to a single file.
// The File interface is the minimum implementation required of the file.
// Directory files should also implement ReadDirFile.
// A file may implement io.ReaderAt or io.Seeker as optimizations.
type File interface {
	Stat() (FileInfo, error)
	Read([]byte) (int, error)
	Close() error
}
```